swift/ql/src/queries/Security/CWE-079/UnsafeWebViewFetch.ql

```diff
@@ -0,0 +1,143 @@
+/**
+ * @name Unsafe WebView fetch
+ * @description Fetching data in a WebView without restricting the base URL may allow an attacker to access sensitive local data, or enable cross-site scripting attack.
+ * @kind path-problem
+ * @problem.severity warning
+ * @security-severity 6.1
+ * @precision high
+ * @id swift/unsafe-webview-fetch
+ * @tags security
+ *       external/cwe/cwe-079
+ *       external/cwe/cwe-095
+ *       external/cwe/cwe-749
+ */
+
+import swift
+import codeql.swift.dataflow.DataFlow
+import codeql.swift.dataflow.TaintTracking
+import codeql.swift.dataflow.FlowSources
+import DataFlow::PathGraph
+import codeql.swift.frameworks.StandardLibrary.String
+
+/**
+ * A taint source that is `String(contentsOf:)`.
+ * TODO: this shouldn't be needed when `StringSource` in `String.qll` is working.
+ */
+class StringContentsOfURLSource extends RemoteFlowSource {
```

Comment:
Oh, right. I see why you need this class. This is essentially the same issue as in https://github.com/github/codeql/pull/9673#discussion_r905171464. I have a fix for our MaD library so that we can delete this class, I'll put up a PR that fixes this once we've merged this PR.

Reply:
Fantastic!  I was going to create an issue for this, but it sounds like you're already on it and that would just be a duplicate, so I'll l eave it in your hands.